pyne/alara.py

```diff
@@ -230,13 +248,27 @@ def photon_source_hdf5_to_mesh(mesh, filename, tags):
             matched_data = h5f.root.data.read_where(
                 "(nuc == '{0}') & (time == '{1}')".format(nuc, cond[1]))
 
-        idx = 0
-        for i, _, ve in mesh:
-            if matched_data[idx][0] == i:
-                tag_handles[tags[cond]][ve] = matched_data[idx][3]
-                idx += 1
-            else:
-                tag_handles[tags[cond]][ve] = [0] * num_e_groups
+        if not sub_voxel:
+            idx = 0
+            for i, _, ve in mesh:
+                if matched_data[idx][0] == i:
+                    tag_handles[tags[cond]][ve] = matched_data[idx][3]
+                    idx += 1
+                else:
+                    tag_handles[tags[cond]][ve] = [0] * num_e_groups
+        else:
+            temp_mesh_data = np.empty(
+                shape=(num_vol_elements, num_e_groups * max_num_cells))
+            for i in range(num_vol_elements):
+                temp_mesh_data[i] = [0.0] * num_e_groups * max_num_cells
```

Comment:
Isn't this already accomplished by creating this as `np.empty()`

Reply:
When we create this as `np.empty()`, it actually contains numbers like  `-2.41598391e-039,   6.91406097e-310,   6.91404841e-310,`, and there is some number like `-1.85792002e+163`, this will make unexpected error. So, I think set them all to be `0.0` is not a bad choice. 

Also, I will use `numpy.fill` command to replace the loop.

Reply:
right - maybe it should be `np.zeros` instead?